src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/LockFactory.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+
+namespace Microsoft.AspNetCore.Razor;
+
+internal class LockFactory
```

Comment:
Could this be a struct, since it just wraps a single `ReaderWriterLockSlim` reference?

Reply:
It definitely can be. I didn't think too much about it since the creation of them is small and long lasting... happy to change